flink-table/flink-table-api-java/src/test/java/org/apache/flink/table/catalog/GenericInMemoryCatalogTest.java

```diff
@@ -562,6 +568,158 @@ public void testRenameView() throws Exception {
 		catalog.dropTable(viewPath2, false);
 	}
 
+	// ------ functions ------
+
+	@Test
+	public void testCreateFunction() throws Exception {
+		catalog.createDatabase(db1, createDb(), false);
+
+		assertFalse(catalog.functionExists(path1));
+
+		catalog.createFunction(path1, createFunction(), false);
+
+		assertTrue(catalog.functionExists(path1));
+
+		catalog.dropFunction(path1, false);
+		catalog.dropDatabase(db1, false);
+	}
+
+	@Test
+	public void testCreateFunction_DatabaseNotExistException() throws Exception {
+		assertFalse(catalog.databaseExists(db1));
+
+		exception.expect(DatabaseNotExistException.class);
+		exception.expectMessage("Database db1 does not exist in Catalog");
```

Comment:
```suggestion
		exception.expectMessage("Database db1 does not exist in catalog");
```

seems no need to have an upper "C". Change the "C" to "c" in DatabaseNotExistException?

Reply:
Did you mean "C" in "Catalog"? Your suggestion might be valid. However, this isn't introduced in this PR and it's consistent with other exceptions. If they are doomed to be changed, I think it's better to do so in a separate JIRA. Thoughts?